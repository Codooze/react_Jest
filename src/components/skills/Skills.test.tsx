import { render, screen } from "@testing-library/react";
import { Skills } from "./skills";

describe("Skills", () => {
  it("should render correctly", () => {
    const { container } = render(<Skills skills={["HTML", "CSS", "JS"]} />);
    expect(container).toMatchSnapshot();
  });

  it("should render the button element", () => {
    const { getByText, getByRole } = render(
      <Skills skills={["HTML", "CSS", "JS"]} />
    );
    const buttonElement = getByText("Login");
    expect(buttonElement).toBeInTheDocument();

    //getByRole
    const buttonElement2 = getByRole("button");
    expect(buttonElement2).toBeInTheDocument();
  });

  //render the skills list
  it("should render the skills list", () => {
    render(<Skills skills={["HTML", "CSS", "JS"]} />);
    const listItems = screen.getAllByRole("listitem");
    expect(listItems).toHaveLength(3);
  });

  //renders button
  it("should render the button element 2", () => {
    render(<Skills skills={["HTML", "CSS", "JS"]} />);
    const buttonElement = screen.getByRole("button", { name: /login/i });

    expect(buttonElement).toBeInTheDocument();
  });

  //start learning button is not rendered
  it("should not render the start learning button", () => {
    render(<Skills skills={["HTML", "CSS", "JS"]} />);
    const buttonElement = screen.queryByRole("button", {
      name: /start learning/i,
    });

    expect(buttonElement).not.toBeInTheDocument();
  });

  //start learning button is eventually rendered
  it("should render the start learning button", async () => {
    render(<Skills skills={["HTML", "CSS", "JS"]} />);
    const buttonElement = await screen.findByRole("button", {
      name: /start learning/i,
    });
    expect(buttonElement).toBeInTheDocument();
  });
});
