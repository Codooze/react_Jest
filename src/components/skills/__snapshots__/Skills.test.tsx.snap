// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills should render correctly 1`] = `
<div>
  <ul>
    <li>
      HTML
    </li>
    <li>
      CSS
    </li>
    <li>
      JS
    </li>
  </ul>
  <button>
    Login
  </button>
</div>
`;
